### PR TITLE
Kernel+Userland: Simplify loading of an ELF interpreter path

### LINKED ISSUE #24762: Kernel: INTERP section out of first 4KiB of ELF leads to validation being failed
How to reproduce:
- Copy `/bin/ls` to `/tmp/ls`
- Run `patchelf --set-interpreter /bin/something /tmp/ls` (using #24761)
- Try to run the copied ELF now. Reverting back to using `/usr/lib/Loader.so` doesn't help as well

```
56.442 [#0 Shell(52:52)]: Found PT_INTERP header (9), but the .interp section was not within the buffer :(
56.442 [#0 Shell(52:52)]: exec(/tmp/ls): File has invalid ELF Program headers
```

### PR DESCRIPTION
**Pull request by supercomputer7**
Fixes #24762.
Paired with #24761, it should be quite easy to test this :)

### CODE REVIEW

File: Kernel/Syscalls/execve.cpp [hunk 1]
```diff
@@ -754,72 +754,82 @@ ErrorOr<RefPtr<OpenFileDescription>> Process::find_elf_interpreter_for_executabl
...
     if (main_executable_requested_stack_size.has_value() && (!minimum_stack_size.has_value() || *minimum_stack_size < *main_executable_requested_stack_size))
         minimum_stack_size = main_executable_requested_stack_size;
 
-    if (!interpreter_path.is_empty()) {
-        dbgln_if(EXEC_DEBUG, "exec({}): Using program interpreter {}", path, interpreter_path);
-        auto interpreter_description = TRY(VirtualFileSystem::the().open(credentials(), interpreter_path, O_EXEC, 0, current_directory()));
-        auto interp_metadata = interpreter_description->metadata();
+    // NOTE: The ELF file might not have any INTERP header, which in such
+    // case we can't do anything and therefore we should just continue
+    // without loading any interpreter.
```
Comment (timschumi) [inline Kernel/Syscalls/execve.cpp:769]: Not a NOTE.

File: Kernel/Syscalls/execve.cpp [hunk 2]
```diff
@@ -747,6 +747,13 @@ static ErrorOr<Vector<NonnullOwnPtr<KString>>> find_shebang_interpreter_for_exec
 
 ErrorOr<RefPtr<OpenFileDescription>> Process::find_elf_interpreter_for_executable(StringView path, Elf_Ehdr const& main_executable_header, size_t main_executable_header_size, size_t file_size, Optional<size_t>& minimum_stack_size)
 {
+    // NOTE: We can't exec an ET_REL, as that's just an object file from the compiler,
```
Comment (timschumi) [inline Kernel/Syscalls/execve.cpp:750]: Not a NOTE.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Kernel/Syscalls/execve.cpp [hunk 3]
```diff
@@ -754,72 +754,82 @@ ErrorOr<RefPtr<OpenFileDescription>> Process::find_elf_interpreter_for_executabl
...
-        Optional<size_t> interpreter_requested_stack_size {};
-        if (!TRY(ELF::validate_program_headers(*elf_header, interp_metadata.size, { first_page, nread }, &interpreter_interpreter_path_builder, &interpreter_requested_stack_size))) {
-            dbgln("exec({}): Interpreter ({}) has invalid ELF Program headers", path, interpreter_path);
-            return ENOEXEC;
-        }
-        auto interpreter_interpreter_path = interpreter_interpreter_path_builder.string_view();
-        if (interpreter_requested_stack_size.has_value() && (!minimum_stack_size.has_value() || *minimum_stack_size < *interpreter_requested_stack_size))
-            minimum_stack_size = interpreter_requested_stack_size;
+    // Validate the program interpreter as a valid elf binary.
+    // If your program interpreter is a #! file or something, it's time to stop playing games :)
```
[the remark this answers is not in the document]
Comment (supercomputer7) [reply Kernel/Syscalls/execve.cpp:799]: I don't know, it's copied from the old code... :sweat_smile: 
I will remove it.

**Review by timschumi — APPROVED** [review]
LGTM, waiting for CI.

**Pull request merged**